Sources/SwiftFormat/Rules/UseShorthandTypeNames.swift

```diff
@@ -47,24 +47,28 @@ public final class UseShorthandTypeNames: SyntaxFormatRule {
 
     switch node.name.text {
     case "Array":
-      guard let typeArgument = genericArgumentList.firstAndOnly else {
+      guard let argument = genericArgumentList.firstAndOnly,
+            case .type(let typeArgument) = argument else {
         newNode = nil
         break
       }
+
       newNode = shorthandArrayType(
-        element: typeArgument.argument,
+        element: typeArgument,
         leadingTrivia: leadingTrivia,
         trailingTrivia: trailingTrivia
       )
 
     case "Dictionary":
-      guard let typeArguments = exactlyTwoChildren(of: genericArgumentList) else {
+      guard let arguments = exactlyTwoChildren(of: genericArgumentList),
+            case .type(let type0Argument) = arguments.0.argument,
+            caes .type(let type1Argument) = arguments.1.argument else {
```

Comment:
Similar here, I think this can be 
 
```suggestion
      guard case (.type(let type0Argument), .type(let type1Argument)) = exactlyTwoChildren(of: genericArgumentList) else {
```

If I’m mistaken, this is good as-is.

Same one more time below.